src/databricks/labs/remorph/cli.py

```diff
@@ -91,53 +92,106 @@ def _verify_workspace_client(ws: WorkspaceClient) -> WorkspaceClient:
 @remorph.command
 def transpile(
     w: WorkspaceClient,
-    transpiler_config_path: str,
-    source_dialect: str,
-    input_source: str,
-    output_folder: str | None,
-    error_file_path: str | None,
-    skip_validation: str,
-    catalog_name: str,
-    schema_name: str,
+    transpiler_config_path: str | None = None,
+    source_dialect: str | None = None,
+    input_source: str | None = None,
+    output_folder: str | None = None,
+    error_file_path: str | None = None,
+    skip_validation: str | None = None,
+    catalog_name: str | None = None,
+    schema_name: str | None = None,
 ):
     """Transpiles source dialect to databricks dialect"""
-    with_user_agent_extra("cmd", "execute-transpile")
     ctx = ApplicationContext(w)
+    logger.debug(f"Application transpiler config: {ctx.transpile_config}")
+    checker = _TranspileConfigChecker(ctx.transpile_config)
+    checker.use_transpiler_config_path(transpiler_config_path)
+    checker.use_source_dialect(source_dialect)
+    checker.use_input_source(input_source)
+    checker.use_output_folder(output_folder)
+    checker.use_error_file_path(error_file_path)
+    checker.use_skip_validation(skip_validation)
+    checker.use_catalog_name(catalog_name)
+    checker.use_schema_name(schema_name)
+    config, engine = checker.check()
+    asyncio.run(_transpile(ctx, config, engine))
+
+
+class _TranspileConfigChecker:
```

Comment:
nit: can we find a better place for this class than cli, I want to keep it lean since this is entry point, Happy to use a singleton for replacing this.